src/types/unfinished_block.py

```diff
@@ -20,7 +20,7 @@ class UnfinishedBlock(Streamable):
     foliage_sub_block: FoliageSubBlock  # Reward chain foliage data
     foliage_block: Optional[FoliageBlock]  # Reward chain foliage data (tx block)
     transactions_info: Optional[TransactionsInfo]  # Reward chain foliage data (tx block additional)
-    transactions_generator: Optional[Program]  # Program that generates transactions
+    transactions_generator: Optional[SerializedProgram]  # Program that generates transactions
```

Comment:
I'm not confident in this change. I don't think `mypy` covers much of the interactions with this class, nor the unit tests (afaict)

Reply:
The reason I made this change is because full blocks are created from this type, and this field is used to initialize the corresponding field in full block

Reply:
Yes if full block is changed, the unfinished block should be changed as well. 
I would check with @Yostra, there are not so many places where Program is used in the node (SpendBundle is one of them though).
It is used more by the wallet and the tests.

Reply:
This is a good place for Matt to look. I can also help re: generator.